### ISSUE TITLE
metadata in ancestor map not inherited by topics in childmap

### ISSUE THREAD

**Issue by jelovirt** [dita-ot/dita-ot#1086]
Converted from [SourceForge issue 2043490](http://sourceforge.net/support/tracker.php?aid=2043490), submitted by julia_m

We are using DITA OT 1.4.1. This issue has to do with the "metadata pushing section" of the OT.

In this link: http://docs.oasis-open.org/dita/v1.1/CS01/archspec/topicmetainheritance.html

It is stated that prodinfo metadata is inherited from maps to topics. My testing confirms that metadata at the map level does not show up in html versions of topics in the map, if the topics belong to a nested map. 

For example, topics A-D belong to a content_unit map that is reused in multiple larger maps. The larger maps are used for different products. According to the spec, topics A-D should inherit prodinfo metadata from larger maps.

That is, in this map:

```
<map>
  ... product info here ...
  <topicref> this topic inherits the metadata info from the map</topicref>
  <topicref href="other-map"> ... topics in this other_map do not inherit metadata from the larger map
</topicref>
</map>
```
- The "other-map" is the content_unit map that is referenced in several larger product maps.
- If topics in the "other-map" do not contain any prodinfo metadata (prodname, version and component), and the "other-map" does not contain any prodinfo metadata, when this map is referenced in the larger map, the topics from the other_map do not inherit prodinfo from the larger map.
- However, those topics that belong to the larger map directly, inherit metadata from the map (if they don't have their own metadata specified), which is according to the spec.

To summarize, smaller maps and their topics should be made metadata-agnostic and inherit metadata from parent maps. 

Thank you,

Julia

**Comment by c094728** [comment]
I had this same issue.  I had to make the re-order change in dc-preprocess target of
org.dita-community.preprocess-extension instead of org.dita.base as described in this ticket to make it work.
